### PR TITLE
Add a new version in the ReplicaTokenSerde class

### PR DESCRIPTION
**Pull request by ankagrawal**
Add a new version in the ReplicaTokenSerde to account for persisting replicatype.
Write tests for DiskTokenPersistor.
Note that this fixes a bug in commit 9487e8be3e7db9da2d994aa70f273b739665235f related to deserializing existing replica token files with the new replca token serialization format.

### CODE REVIEW

File: ambry-server/src/integration-test/java/com.github.ambry.server/ServerTestUtil.java [hunk 1]
```diff
@@ -2003,7 +2003,7 @@ private static void checkReplicaTokens(MockClusterMap clusterMap, DataNodeId dat
           DataInputStream dataInputStream = new DataInputStream(crcStream);
           try {
             short version = dataInputStream.readShort();
-            assertEquals(0, version);
+            assertEquals(1, version);
 
             System.out.println("setToCheck" + setToCheck.size());
```
Comment (jsjtzyy) [inline ambry-server/src/integration-test/java/com.github.ambry.server/ServerTestUtil.java]: remove this?

File: ambry-api/src/main/java/com.github.ambry/config/ReplicationConfig.java [hunk 2]
```diff
@@ -27,7 +27,7 @@
   /**
    * The factory class the replicatio uses to create cloud token
```
Comment (jsjtzyy) [inline ambry-api/src/main/java/com.github.ambry/config/ReplicationConfig.java]: minor: typo `replication`

File: ambry-server/src/integration-test/java/com.github.ambry.server/ServerTestUtil.java (hunk 1 above)
Comment (ankagrawal) [reply ambry-server/src/integration-test/java/com.github.ambry.server/ServerTestUtil.java]: this was from before, but fixed nonetheless!

File: ambry-api/src/main/java/com.github.ambry/config/ReplicationConfig.java (hunk 2 above)
Comment (ankagrawal) [reply ambry-api/src/main/java/com.github.ambry/config/ReplicationConfig.java]: fixed!

**Pull request merged**